### PR TITLE
Delete corrupted snapshots if valid snapshot is available

### LINKED ISSUE #7531: Delete corrupted snapshots if a newer valid snapshot is available
**Description**

When committing a new valid snapshot, we currently only delete the previous valid snapshot. To mitigate having multiple older snapshots, on start up, we load all valid snapshots, then keep the newest and delete the others. However, if a snapshot is not valid on start up (e.g. corrupted), then it won't be deleted, and neither will it be deleted when a new valid one is committed. This means there is garbage data taking up space for no reason.

We should clean up corrupted snapshots when committing a new valid snapshot, _and_ on startup _iff_ there is a newer valid snapshot.

### PR DESCRIPTION
**Pull request by lenaschoenburg**
## Description

Previously we only deleted corrupt snapshots in the case of a missing checksum file. Now, if there is a valid snapshot, we delete everything else in the snapshot directory.
This ensures that we don't accumulate corrupted snapshots. It also replaces the logic that deleted older, valid snapshots.

## Related issues

closes #7531

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java [hunk 1]
```diff
@@ -111,38 +109,48 @@ protected void onActorClosing() {
...
 
+  private void cleanupSnapshotDirectory(
+      final Path snapshotDirectory, final FileBasedSnapshot latestPersistedSnapshot)
+      throws IOException {
+    final var latestChecksumFile = latestPersistedSnapshot.getChecksumFile();
+    final var latestDirectory = latestPersistedSnapshot.getDirectory();
+    try (final var paths =
+        Files.newDirectoryStream(
+            snapshotDirectory, p -> !p.equals(latestDirectory) && !p.equals(latestChecksumFile))) {
+      LOGGER.debug("Deleting snapshots other than {}", latestPersistedSnapshot);
```
Comment (deepthidevaki) [inline snapshot/src/main/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStore.java:142]:
```suggestion
      LOGGER.debug("Deleting snapshots other than {}", latestPersistedSnapshot.getId());
```

File: snapshot/src/test/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStoreTest.java [hunk 2]
```diff
@@ -128,6 +128,66 @@ public void shouldDeleteSnapshotWithoutChecksumFile() throws IOException {
...
+
+    final var newerSnapshot =
+        (FileBasedSnapshot) takeTransientSnapshot(2, snapshotStore).persist().join();
+
+    // when
+    snapshotStore.close();
+    snapshotStore = createStore(snapshotsDir, pendingSnapshotsDir);
+
+    // then
+    assertThat(snapshotStore.getLatestSnapshot()).hasValue(newerSnapshot);
```
Comment (deepthidevaki) [inline snapshot/src/test/java/io/camunda/zeebe/snapshots/impl/FileBasedSnapshotStoreTest.java:147]: For completeness, I would also suggest to verify that the valid snapshot is not deleted.
```
    assertThat(newerSnapshot.getDirectory()).exists();
    assertThat(newerSnapshot.getChecksumFile()).exists();
```

**Review by deepthidevaki — CHANGES_REQUESTED** [review]
:rocket: awesome!

I have only some minor comments. Please see below.
In addition, it would be good to add one more test to verify the corrupted snapshots are not deleted if there is no other valid snapshot. What do you think?

**Comment by oleschoenburg** [comment]
@deepthidevaki Thanks for the review! I agree with you and added an additional test case that confirms that a corrupt snapshot is not deleted if there is no valid snapshot available.

**Review by deepthidevaki — APPROVED** [review]
Great! :tada:

**Pull request merged**